opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -125,12 +150,9 @@ impl<T: Number<T>> Sum<T> {
         s_data.is_monotonic = self.monotonic;
         s_data.data_points.clear();
 
-        let mut values = match self.value_map.values.write() {
-            Ok(v) => v,
-            Err(_) => return (0, None),
-        };
-
-        let n = values.len() + 1;
+        // Max number of data points need to account for the special casing
+        // of the no attribute value + overflow attribute.
+        let n = self.value_map.count.load(Ordering::Relaxed) + 2;
```

Comment:
Let's start with `SeqCst` and then investigate if we could relax the ordering without affecting correctness?

```suggestion
        let n = self.value_map.count.load(Ordering::SeqCst) + 2;
```